src/app/services/geo-data.service.ts

```diff
@@ -298,10 +308,167 @@ export class GeoDataService {
     });
   }
 
+  deleteTileServer(projectId: number, tileServerId: number) {
+    this.http
+      .delete(environment.apiUrl + `/projects/${projectId}/tile-servers/${tileServerId}/`)
+      .subscribe((resp) => {
+        this.tileServers$.pipe(
+          take(1),
+          map((items: Array<TileServer>) =>
+            items.filter((item: TileServer) =>
+              item.id !== tileServerId)),
+        ).subscribe((results) =>  {
+          this._tileServers.next(results);
+          this.notificationsService.showSuccessToast('Tile layer deleted!');
+        });
+      }, (error => {
+        console.log(error);
+        this.notificationsService.showErrorToast('Tile layer could not be deleted!');
+      }));
+  }
+
+  public updateTileServers(projectId: number, tileServers: Array<TileServer>) {
+    this._tileServers.next(tileServers);
+    this._dirtyTileOptions.next(true);
+  }
+
+  public updateTileServer(projectId: number, tileServer: TileServer): void {
+    this.tileServers$.pipe(
+      take(1),
+      map((tss: Array<TileServer>) =>
+        tss.map((ts: TileServer) =>
+          ts.id == tileServer.id ? tileServer : ts)),
+    ).subscribe((results) =>  {
+      this._tileServers.next(results);
+      this._dirtyTileOptions.next(true);
+    });
+  }
+
+  public saveTileServers(projectId: number, tileServers: Array<TileServer>): void {
+    if (this._dirtyTileOptions.value) {
```

Comment:
I'm not following the if/else branch here; can you explain this. thanks!

Reply:
I was thinking about branching it so that you won't get the toaster on each save operation (specifically when using it in `addTileServer`).

Reply:
Would this be better? 2c1aaf0